### PR TITLE
NXBT-3829: Automate jira release

### PR DESCRIPTION
**Pull request by kevinleturc**
This change come with a tag renaming to follow the vX.Y.Z convention.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -120,8 +121,35 @@ pipeline {
         container('base') {
           nxWithGitHubStatus(context: 'release', message: 'Release') {
             script {
-              // tag the version - nuxeo-helm-chart doesn't follow the v1.2.3 convention
-              nxGit.commitTagPush(tag: env.VERSION)
```
Comment (ataillefer) [inline Jenkinsfile:124]: Does this mean we are switching to v3.x.y?

File: Jenkinsfile [hunk 2]
```diff
@@ -120,8 +121,35 @@ pipeline {
...
+            def jiraVersion = [
+                project: 'NXP',
+                name: jiraVersionName,
+                releaseDate: LocalDate.now().format(DateTimeFormatter.ISO_LOCAL_DATE),
+                released: true,
+            ]
+            nxJira.newVersion(version: jiraVersion)
+            // find Jira tickets included in this release and update them
+            def jiraTickets = nxJira.jqlSearch(jql: "project = NXP and fixVersion = '${JIRA_NUXEO_CHART_MOVING_VERSION}'")
+            def previousVersion = sh(returnStdout: true, script: "perl -pe 's/\\b(\\d+)(?=\\D*\$)/\$1-1/e' <<< ${VERSION}").trim()
```
Comment (ataillefer) [inline Jenkinsfile:148]: Do we want to take care of a new minor or major version, aka 3.2.0 or 4.0.0?

**Review by ataillefer — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (kevinleturc) [reply Jenkinsfile:124]: Yes it is.
I also renamed all previous tags to this pattern.
I didn't see any impact on such, we're just losing the tag date, but not a big deal as we have the tagged commit date.

File: Jenkinsfile (hunk 2 above)
Comment (kevinleturc) [reply Jenkinsfile:148]: If I remember correctly, the perl script will produce a wrong version, like `4.0.-1`, but it is caught by the git change log method that fallback on a full retrieval of history.
In the end, if nothing fail, this code is more or less optimization for git commands.
🤞

**Pull request merged**